### PR TITLE
Fixes for errors in the "array_ptr" interfaces to some libc functions declared in checkedc_extensions.h, and removal of the conditional in the snprintf declaration in stdio_checked.h

### PR DESCRIPTION
**Pull request by sulekhark**
This PR contains fixes for issues #449 and #450. They are, in brief:
1) link-time errors caused by declaring `strncmp_array_ptr` as `extern inline`,
2) missing definition for `snprintf_array_ptr`, and
3) inconsistency between the second argument of `snprintf` (i.e. the argument `n`) and the bounds declaration for the first argument of `snprintf` (i.e., the argument `s`) when n == 0.

### CODE REVIEW

File: tests/dynamic_checking/checked_headers/ext.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// The following lines are for the LLVM test harness:
+//
+// RUN: %clang -o %t.exe %s %checkedc_target_flags
+// RUN: %checkedc_rununder %t.exe
+
+// expected-no-diagnostics
+
+#pragma CHECKED_SCOPE on
+
+#include <stddef.h>
```
Comment (mgrang) [inline tests/dynamic_checking/checked_headers/ext.c:12]: Can these headers be listed alphabetically?

File: include/checkedc_extensions.h [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 #include <stdlib_checked.h>
 #include <string_checked.h>
+#include <stdarg.h>
```
Comment (mgrang) [inline include/checkedc_extensions.h:9]: Can these headers be listed alphabetically?

**Review by mgrang — APPROVED** [review]
LGTM.

**Comment by mgrang** [comment]
Could you please add a title describing the changes?

**Comment by mattmccutchen-cci** [comment]
Ideally, the new PR title would encompass the removal of the conditional from `snprintf` (#450) as well as the `checkedc_extensions.h` changes (#449).  Sulekha, would you be willing to change it?

**Review by mgrang — APPROVED** [review]
LGTM. Thanks!

**Comment by sulekhark** [comment]
> Ideally, the new PR title would encompass the removal of the conditional from `snprintf` (#450) as well as the `checkedc_extensions.h` changes (#449). Sulekha, would you be willing to change it?

Yes, of course.

**Pull request merged**